watertap/tools/parameter_sweep/parameter_sweep_differential.py

```diff
@@ -381,21 +375,73 @@ def _do_param_sweep(self, model, sweep_params, outputs, local_values):
 
     def parameter_sweep(
         self,
-        model,
-        sweep_params,
-        outputs=None,
+        build_model,
+        build_sweep_params,
+        build_outputs=None,
+        build_outputs_kwargs=dict(),
         num_samples=None,
         seed=None,
+        build_model_kwargs=dict(),
+        build_sweep_params_kwargs=dict(),
```

Comment:
```suggestion
        build_outputs_kwargs=None,
        num_samples=None,
        seed=None,
        build_model_kwargs=None,
        build_sweep_params_kwargs=None,
```

We also need to add a conversion to `dict` for `build_outputs_kwargs` if it is `None`.

We never use mutable default arguments in Python, see, e.g., https://stackoverflow.com/questions/1132941/least-astonishment-and-the-mutable-default-argument